### PR TITLE
chore: show errors on Login page from backend

### PR DESCRIPTION
**Pull request by yquansah**
This PR makes a change to show login errors on the UI that come from the backend. It does the following:
- Adds a `LoginWrapper` component, so that the actual context for errors can be accessed
- Uses that `LoginWrapper` component in the `Login` component

Completes FLI-539

Screenshot 2023-08-15 at 12 47 23 PM

From React documentation:

> useContext() call in a component is not affected by providers returned from the same component. The corresponding <Context.Provider> **needs to be above** the component doing the useContext() call.

### CODE REVIEW

File: ui/src/app/auth/Login.tsx [hunk 1]
```diff
@@ -1,183 +1,12 @@
...
-                  </div>
-                )}
-              </div>
-            </div>
-          </div>
-        </main>
-      </div>
-    </>
+    <NotificationProvider>
+      <LoginWrapper />
```
Comment (markphelps) [inline ui/src/app/auth/Login.tsx:8]: I would just do all this in a single file IMO no need for `LoginWrapper`, could just do like we do in `Layout.tsx`:

```jsx
function InnerLogin() {
 // the contents of LoginWrapper here
}

export default function Login() {
return (
  <NotificationProvider>
    <InnerLogin />
    <ErrorNotification />
  </NotificationProvider>
)
}
```

File: ui/src/components/notifications/ErrorNotification.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ export default function ErrorNotification() {
 
   return (
     <Transition show={error !== null}>
-      <div className="max-w-s fixed bottom-0 right-2 z-50 m-4">
+      <div className="max-w-s fixed bottom-0 right-2 z-50 m-4 w-1/3">
```
Comment (markphelps) [inline ui/src/components/notifications/ErrorNotification.tsx:10]: does this look ok on other pages? that cause errors?

Comment (markphelps) [reply ui/src/components/notifications/ErrorNotification.tsx:10]: actually, what if instead we do `max-w-1/3`? that way for notifications with shorter error messages it still will fit those without being too large? would that work here?

Comment (yquansah) [reply ui/src/components/notifications/ErrorNotification.tsx:10]: @markphelps That works. I tested this on other errors yes! but yeah i think having it `max` makes sense.

Comment (yquansah) [reply ui/src/components/notifications/ErrorNotification.tsx:10]: @markphelps Hmm i tried this `max-w-1/3` and I am not sure if that is a proper setting with Tailwind. It seemed as though this error displayed above was one of the more larger ones throughout the app, so I picked a safe value.

**Review by markphelps — APPROVED** [review]
nice!

**Pull request merged**